### ISSUE TITLE
[FEATURE] VS Code extension

### ISSUE THREAD

**Issue by utybo** [tapjs/tapjs#980]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you read the `CONTRIBUTING` guide on posting issues, and `CODE_OF_CONDUCT`?

- [X] yes I read the things

### Description

I recently migrated my new project from Jest to node-tap. I could not get Jest working with TypeScript and ESM, and was fed up with nothing ever documented working. I dropped Jest for node-tap and everything worked out-of-the-box, which is great! Thank you for making my life much easier 🙏🏻 

The main thing that I'm missing from Jest is a VS Code extension that populates the "Test" panel in VS Code. This makes browsing through and running specific tests super easy (both from the Test panel and from the files directly). It would be really cool to also have a code coverage integration in VS Code.

- Does such an extension already exist?
- Is there a more general TAP extension that could work with node-tap in VS Code?
- Are there plans on developing this?
- Otherwise, if someone has already tried it, what are the current roadblocks for implementing such an extension?

I'd be willing to try creating such an extension, though I don't have any experience with it.

### Example

Main features I can think of, in order of priority

- Implement VS Code's Testing API with node-tap: https://code.visualstudio.com/api/extension-guides/testing
  - Allows running tests by clicking in the gutter
- Debugging tests
- Make it Just Work™️ as much as possible, just like node-tap
- Fully optional configuration (choose which command or script is used for running node-tap, that sort of stuff)
- Code coverage gutters, either by being compatible with the "Coverage Gutters" plugin, or copying how they do it.

I think a lot of inspiration can be taken from the [Jest plugin](https://github.com/jest-community/vscode-jest) for most of these.

**Comment by isaacs** [comment]
Yes, i definitely love this idea, but im not sure im the best one to build it, since i dont use vscode and thus dont have much of a sense for what would feel "right" there. (I use nvim, which probably explains why tap is so optimized for the cli, and not much else ;)

Do you want to give it a try? I'd be happy to answer any questions or offer help if i can.

**Comment by utybo** [comment]
> Do you want to give it a try? I'd be happy to answer any questions or offer help if i can.

Sure! I'll give it a shot.

**Comment by justintime4tea** [comment]
> > Do you want to give it a try? I'd be happy to answer any questions or offer help if i can.
> 
> Sure! I'll give it a shot.

I noticed you mentioned working on this in a couple of different issues but I couldn't track down a repo for the node-tap/tapjs VS Code extension. Do you have it hosted somewhere? I'm also interested in this and would love to take a look. Not sure how much time I'd have to contribute but I would sure love to help make this happen somehow.

**Comment by utybo** [comment]
> I noticed you mentioned working on this in a couple of different issues but I couldn't track down a repo for the node-tap/tapjs VS Code extension. Do you have it hosted somewhere? I'm also interested in this and would love to take a look. Not sure how much time I'd have to contribute but I would sure love to help make this happen somehow.

Unfortunately I stopped working with TapJS shortly after deciding to give making the extension a try - I decided to use another language for the project I wanted to do instead of JS. Sorry :(
